includes/Classifai/Providers/Azure/TextToSpeech.php

```diff
@@ -0,0 +1,553 @@
+			);
+			// phpcs:enable
+		}
+
+		return $options;
+	}
+
+	/**
+	 * Provides debug information related to the provider.
+	 *
+	 * @param null|array $settings   Settings array. If empty, settings will be retrieved.
+	 * @param boolean    $configured Whether the provider is correctly configured. If null, the option will be retrieved.
+	 * @return array Keyed array of debug information.
+	 */
+	public function get_provider_debug_information( $settings = null, $configured = null ) {
+		if ( is_null( $settings ) ) {
+			$settings = $this->sanitize_settings( $this->get_settings() );
+		}
+
+		$authenticated = 1 === intval( $settings['authenticated'] ?? 0 );
+
+		return [
+			__( 'Authenticated', 'classifai' )  => $authenticated ? __( 'Yes', 'classifai' ) : __( 'No', 'classifai' ),
+			__( 'API URL', 'classifai' )        => $settings['url'] ?? '',
+		];
+	}
+
+	/**
+	 * Returns the default settings.
+	 */
+	private function get_default_settings() {
+		return [
+			'url'           => '',
+			'api_key'       => '',
+			'voice'         => '',
+			'authenticated' => false,
+		];
+	}
+
+	/**
+	 * Add `classifai_synthesize_speech` to rest API for view/edit.
+	 */
+	public function add_synthesize_speech_meta_to_rest_api() {
+		$settings             = \Classifai\get_azure_text_to_speech_settings();
+		$supported_post_types = isset( $settings['post_types'] ) ? $settings['post_types'] : array();
+
+		$supported_post_types = array_keys(
+			array_filter(
+				$supported_post_types,
+				function( $post_type ) {
+					return ! is_null( $post_type );
+				}
+			)
+		);
+
+		register_rest_field(
+			$supported_post_types,
+			'classifai_synthesize_speech',
+			array(
+				'get_callback'    => function( $object ) {
+					$process_content = get_post_meta( $object['id'], self::SYNTHESIZE_SPEECH_KEY, true );
+					return ( 'no' === $process_content ) ? 'no' : 'yes';
+				},
+				'update_callback' => function ( $value, $object ) {
+					$value = ( 'no' === $value ) ? 'no' : 'yes';
+					return update_post_meta( $object->ID, self::SYNTHESIZE_SPEECH_KEY, $value );
+				},
+				'schema'          => [
+					'type'    => 'string',
+					'context' => [ 'view', 'edit' ],
+				],
+			)
+		);
+
+		register_rest_field(
+			$supported_post_types,
+			'classifai_post_audio_id',
+			array(
+				'get_callback' => function( $object ) {
+					$post_audio_id = get_post_meta( $object['id'], self::AUDIO_ID_KEY, true );
+					return (int) $post_audio_id;
+				},
+				'schema'       => [
+					'type'    => 'integer',
+					'context' => [ 'view', 'edit' ],
+				],
+			)
+		);
+	}
+
+	/**
+	 * Adds audio controls to the post that has speech sythesis enabled.
+	 *
+	 * @param string $content Post content.
+	 * @return string
+	 */
+	public function render_post_audio_controls( $content ) {
+		global $post;
+
+		if ( ! $post ) {
+			return $content;
+		}
+
+		if ( apply_filters( 'classifai_disable_post_to_audio_block', false, $post->ID ) ) {
+			return $content;
+		}
+
+		wp_enqueue_script(
```

Comment:
Should probably move these enqueues after the check for an audio attachment ID, so we're not loading these assets if we don't actually have audio to play